host-controller/src/main/java/org/jboss/as/host/controller/DomainModelControllerService.java

```diff
@@ -640,11 +641,53 @@ protected void boot(final BootContext context) throws ConfigurationPersistenceEx
 
             // handler for domain server auth.
             DomainManagedServerCallbackHandler.install(serviceTarget);
+
             // Parse the host.xml and invoke all the ops. The ops should rollback on any Stage.RUNTIME failure
             List<ModelNode> hostBootOps = hostControllerConfigurationPersister.load();
 
+            ConfigurationFile.InteractionPolicy configPolicy = environment.getHostConfigurationFile().getInteractionPolicy();
+            if (!environment.getRunningModeControl().isReloaded() && (configPolicy == ConfigurationFile.InteractionPolicy.NEW || configPolicy == ConfigurationFile.InteractionPolicy.DISCARD)) {
+
+                // minimum needed to boot an embedded HC from an empty host.xml
+                ModelNode baseOp;
+
+                baseOp = new ModelNode();
+                baseOp.get("operation").set("register-host-model");
+                baseOp.get("name").set("master");
+                baseOp.get("address").addEmptyList();
+                hostBootOps.add(baseOp);
+
+                baseOp = new ModelNode();
+                baseOp.get("operation").set("write-attribute");
+                baseOp.get("name").set("name");
+                baseOp.get("value").set("master");
+                baseOp.get("address").set(PathAddress.pathAddress("host", "master").toModelNode());
+                hostBootOps.add(baseOp);
+
+                // mgmt extension
+                baseOp = new ModelNode();
+                baseOp.get("operation").set("add");
+                baseOp.get("extension").set("name");
+                baseOp.get("address").set(PathAddress.pathAddress("host", "master").append(
+                        PathAddress.pathAddress("extension", "org.wildfly.extension.core-management")).toModelNode());
+                hostBootOps.add(baseOp);
+
+                // mgmt subsystem
+                baseOp = new ModelNode();
+                baseOp.get("operation").set("add");
+                baseOp.get("address").set(PathAddress.pathAddress("host", "master").
+                        append(PathAddress.pathAddress("subsystem", "core-management")).toModelNode());
+                hostBootOps.add(baseOp);
```

Comment:
Why is this subsystem and extension needed?

Reply:
This is going back a while, but as I recall not having it present would result in the CLI being unable to connect or something similar. It's probably worth a retest at this point and see if it's still behaving in the same way.

If it doesn't work with this present, would you consider that a bug? The specific use case here is booting with empty configs, host name set to 'master' and assumed to be domain controller.